src/components/Modal/TodoCardModal/TodoCardModal.tsx

```diff
@@ -28,7 +28,11 @@ export default function TodoCardModal({ card, column, onClick }: TodoCardModalPr
 
   useEffect(() => {
     refetch();
-  }, [refetch]);
+
+    return () => {
+      queryClient.cancelQueries({ queryKey: ['comments', card.id] });
```

Comment:
cancel이랑 refetch 둘이 동시에 필요한가요???

Reply:
네! 멘토님 피드백대로 언마운트될 때 쿼리가 취소되도록 구현할려고 클린업 함수로 넣어두었습니다!

Reply:
refetch 빼고 cancel만 넣어도 잘 되는 것 같아요!
열기 전에 refetch하거나, 닫기전에 cancel해서 다음에 열 때 refetch하도록 하면 충분할 것 같아서 물어봤어요!

Reply:
클린업만 적용해도 되겠군요..! 수정해보겠습니다
감사합니다 :)

Reply:
invalidateQueries 써도 잘 되는 것 같아요!!
```The cancelQueries method can be used to cancel outgoing queries based on their query keys or any other functionally accessible property/state of the query.
This is most useful when performing optimistic updates since you will likely need to cancel any outgoing query refetches so they don't clobber your optimistic update when they resolve.```
제가 cancel은 또 무슨 다른 기능이 있는지는 모르겠는데, cancel을 고르신 이유가 있나요???

Reply:
cancel은 쿼리 자체를 멈추는 거여서, 언마운트 때 적용하기에 적합하다고 생각했어요! (마운트 때->refetch, 언마운트될 때 -> 클린업으로 댓글 쿼리 취소하기 로직으로 생각했어요)

> 제가 cancel은 또 무슨 다른 기능이 있는지는 모르겠는데, cancel을 고르신 이유가 있나요???

맞아요 invalidate랑 큰 차이가 있는지는 저도 잘 모르겠어요..!! 이런 상황에선 뭘 써야 할지 고민이네요

Reply:
optimistic update 에 좋다네요!!
optimistic update [관련글1](https://velog.io/@mskwon/react-query-cancel-queries), [관련글2](https://sonblog.vercel.app/blogs/blog/react/optimistic-updates)
아마 post, put, delete 등을 할 때 쓰는 것 같은데.. 우리는 get하는 부분이니 어느 것 쓰나 별로 상관 없을 것 같네요!!

Reply:
wow 아티클까지 감사합니다!!! 이 부분은 또 멘토님께 여쭤보죠 :)
일단 이대로 픽스하고 대양님께 자료 넘겨드릴게요!